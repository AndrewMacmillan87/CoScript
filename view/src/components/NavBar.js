import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";
import Button from "./buttons/Button";

import { getUserMessages } from "../redux/auxiliary/user-messages";

// Redux
import { connect } from "react-redux";
import { getMostRecentProject } from "../redux/actions/documents";

import PropTypes from "prop-types";

import "./navbar.css";

export class NavBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            visible: false,
            unread: false,
        };
        this.handleClick = this.handleClick.bind(this);
    }

    /**
     * Checks to see if the user has any unread messages
     * in their personal profile message inbox and sets
     * the 'unread' state property to true if they have.
     * This displays a small icon to the user to indicate
     * they have unread messages
     */
    async componentDidMount() {
        const userMessages = await getUserMessages();
        for (let message of userMessages) {
            if (!message.read) {
                this.setState({
                    unread: true,
                });
            }
        }
    }

    /**
     * Sets the 'visible' state property so that the
     * dropdown menu can be revealed and hidden by
     * clicking the link
     */
    handleClick() {
        this.setState((state) => ({
            visible: !state.visible,
        }));
    }

    render() {
        const { unread } = this.state;
        const { user, loading, history } = this.props;

        const menuIcon = require("../img/menu.svg");
        const spinner = require("../img/loading.svg");
        const isLoggedIn = user && user.isLoggedIn | false;

        const loggedIn = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                        <Link
                            to='#'
                            onClick={async () => {
                                await this.props.getMostRecentProject();
                                const { projectData } = this.props;
                                projectData
                                    ? history.push(`/editor${projectData.url}`)
                                    : history.push("/editor");
                            }}>
                            Workspace
                        </Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    {user && !loading ? (
                        <div className='logged-in-user-container'>
                            <p className='user-username'>
                                {user.profile.displayName}
                            </p>
                            <button
                                className='desktop-icon-link'
                                onClick={this.handleClick}>
                                {unread ? (
                                    <div className='avatar-unread-msg-container'>
                                        <div className='avatar-unread-msg-icon'></div>
                                        <img
                                            className='user-gravatar'
                                            src={user.avatar}
                                            alt='user gravatar'
                                        />
                                    </div>
                                ) : (
                                    <img
                                        className='user-gravatar'
                                        src={user.avatar}
                                        alt='user gravatar'
                                    />
                                )}
                            </button>
                        </div>
                    ) : (
                        <img
                            className='user-spinner'
                            src={spinner}
                            alt='loading spinner icon'
                        />
                    )}
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                    <Link
                        to='#'
                        onClick={async () => {
                            await this.props.getMostRecentProject();
                            const { projectData } = this.props;
                            projectData
                                ? history.push(`/editor${projectData.url}`)
                                : history.push("/editor");
                        }}>
                        <p className='dropdown-link'>Workspace</p>
                    </Link>
                    <Link to='/profile'>
                        <p className='dropdown-link'>My Profile</p>
                    </Link>
                    <a
                        onClick={() => {
                            history["from"] = "logout";
                            history.push("/redirect");
                        }}
                        href=' '>
                        <p className='dropdown-link'>Logout</p>
                    </a>
                </div>
            </Fragment>
        );

        const visiting = (
            <Fragment>
                <div className='header'>
                    <div className='logo-container'>
                        <Link to='/'>
                            <p className='logo'>CoScript</p>
                        </Link>
                    </div>
                    <div className='nav-links'>
                        <Link to='/projects'>Active Projects</Link>
                        <Link to='/publications'>Publications</Link>
                    </div>
                </div>
                <div className='nav-btns'>
                    <div>
                        <Link to='/register'>
                            <Button classes='btn' value='Register' />
                        </Link>
                    </div>
                    <div>
                        <Link to='/login'>
                            <Button classes='grn-btn' value='Login' />
                        </Link>
                    </div>
                </div>
                <button
                    className='dropdown-icon-link'
                    onClick={this.handleClick}>
                    <img
                        className='dropdown-icon'
                        src={menuIcon}
                        alt='Burger menu icon'
                    />
                </button>
                <div
                    id='dropdown-nav'
                    className={
                        this.state.visible
                            ? "dropdown-menu dropdown-content display"
                            : "dropdown-menu dropdown-content"
                    }>
                    <Link to='/register'>
                        <p className='dropdown-link'>Register</p>
                    </Link>
                    <Link to='/login'>
                        <p className='dropdown-link'>Login</p>
                    </Link>
                    <Link to='/projects'>
                        <p className='dropdown-link'>Active Projects</p>
                    </Link>
                    <Link to='/publications'>
                        <p className='dropdown-link'>Publications</p>
                    </Link>
                </div>
            </Fragment>
        );

        return (
            <nav className='navbar'>
                {!loading && (
                    <Fragment>{!isLoggedIn ? visiting : loggedIn}</Fragment>
                )}
            </nav>
        );
    }
}

NavBar.propTypes = {
    user: PropTypes.object,
    loading: PropTypes.bool,
    getMostRecentProject: PropTypes.func.isRequired,
};

const setProps = (state) => ({
    user: state.auth.user,
    loading: state.auth.loading,
    projectData: state.documents.projectData,
});

export default connect(setProps, {
    getMostRecentProject,
})(withRouter(NavBar));
