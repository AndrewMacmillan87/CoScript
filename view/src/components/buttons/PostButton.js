import React from "react";
import PropTypes from "prop-types";

import "./buttons.css";

const PostButton = ({ onClick, classes, value }) => {
    return (
        <input
            onClick={onClick}
            className={classes}
            type='submit'
            value={value}
        />
    );
};

PostButton.defaultProps = {
    value: "Verify"
};

PostButton.propTypes = {
    onClick: PropTypes.func,
    classes: PropTypes.string,
    value: PropTypes.string
};

export default PostButton;
