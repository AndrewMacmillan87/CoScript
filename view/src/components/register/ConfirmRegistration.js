import React from "react";
import { Redirect } from "react-router-dom";
import { Link } from "react-router-dom";

import PostButton from "../buttons/PostButton";

import PropTypes from "prop-types";

const ConfirmRegistration = ({
    isRegistered,
    onChangeConfirm,
    onSubmitConfirm,
    verifyFailed
}) => {
    if (isRegistered) {
        return <Redirect to='/login' />;
    }

    return (
        <main className='confirm-body'>
            <section className='confirm-container'>
                <h1 className='coscript-h1'>CoScript</h1>
                <p className='confirm-text'>
                    Please enter the code we just sent to the email address you
                    signed up with:
                </p>
                <div className='confirm-code-container'>
                    <form
                        className='confirm-code'
                        onSubmit={e => onSubmitConfirm(e)}>
                        <input
                            className='c-code'
                            type='text'
                            name='confirmCode'
                            onChange={e => onChangeConfirm(e)}
                            placeholder='Confirmation code'
                        />
                        <PostButton classes='square-btn' value='Submit' />
                    </form>
                </div>
                <span className='verify-code-failed' ref={verifyFailed}></span>
                <Link to='/'>
                    <p className='home-link'>Back Home</p>
                </Link>
            </section>
        </main>
    );
};

ConfirmRegistration.propTypes = {
    onChangeConfirm: PropTypes.func.isRequired,
    onSubmitConfirm: PropTypes.func.isRequired
};

export default ConfirmRegistration;
