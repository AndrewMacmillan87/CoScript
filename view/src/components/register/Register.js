import React, { useState, useEffect, useRef } from "react";
import { Redirect, withRouter } from "react-router-dom";
import NavBar from "../NavBar";
import RegistrationForm from "./RegistrationForm";
import ConfirmRegistration from "./ConfirmRegistration";

import { sendEmail } from "../../redux/auxiliary/sendEmail";

// Redux
import { connect } from "react-redux";
import { displayAlert } from "../../redux/actions/alertpopup";
import { startRegistration, verifyEmail } from "../../redux/actions/auth";

import PropTypes from "prop-types";

import "./register.css";

export const Register = ({ ...props }) => {
    /**
     * Create references to focus on the emailField on mount and
     * for the code verification field to check for failures
     */
    const emailRef = useRef();
    const verifyRef = useRef();

    /**
     * Set the state for this functional component and its children
     */
    const [email, setEmail] = useState("");
    const [name, setName] = useState("");
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("");
    const [confirmPassword, setConfirmPassword] = useState("");
    const [failed, setFailed] = useState(false);
    const [verifyFailedMsg, setVerifyFailedMsg] = useState("");
    const [confirmCode, setConfirmCode] = useState("");
    const [loading, setLoading] = useState(false);

    /**
     * This effect will set the document title and scroll to the
     * top of the page on mount and unmount. It will also focus
     * on the email field when the component first mounts
     */
    useEffect(() => {
        document.title = "CoScript - Register";
        window.scrollTo(0, 0);
        if (emailRef.current) emailRef.current.focus();
    }, []);

    /**
     * Checks the 'failed' state property to see if a
     * verification code failed to match and sets the
     * 'verifiedFailed' ref textContext to the value of
     * 'verifyFailedMsg' if it did
     */
    useEffect(() => {
        if (failed) {
            if (verifyRef.current)
                verifyRef.current.textContent = verifyFailedMsg;
        }
    }, [failed, verifyFailedMsg]);

    /**
     * Sets the 'loading' state property to true to
     * display a loading spinner while registering
     * the user
     */
    const displaySpinner = () => {
        setLoading(true);
    };

    /**
     * Sets the 'loading' state property to false to
     * remove a loading spinner if registering the
     * user failed
     */
    const removeSpinner = () => {
        setLoading(false);
    };

    /**
     * Sets the <[e.target.name]> state property to
     * the value of the event target
     *
     * @param {Object} e - event object
     */
    const onChange = (e) => {
        const value = e.target.value;
        switch (e.target.name) {
            case "email":
                setEmail(value);
                break;
            case "name":
                setName(value);
                break;
            case "username":
                setUsername(value);
                break;
            case "password":
                setPassword(value);
                break;
            case "confirmPassword":
                setConfirmPassword(value);
                break;
            default:
                return null;
        }
    };

    /**
     * Sets the 'confirmCode' state property to the
     * value of the event target
     *
     * @param {Object} e - event object
     */
    const onChangeConfirm = (e) => {
        setConfirmCode(e.target.value);
    };

    /**
     *First checks to see if the passwords the user entered
     matched, displays a warning alert if not. Then starts 
     the pre-registration process and sends the user an email
     with a a verification code they can use to complete 
     registration. The original version of the code is 
     stored in local storage temporarily **THIS NEEDS TO BE
     CHANGED SO THAT THE CODE IS STORED IN THE DATABASE**
     * 
     * @param {Object} e - event object
     */
    const onSubmit = async (e) => {
        e.preventDefault();
        displaySpinner();
        if (password !== confirmPassword) {
            props.displayAlert("Passwords do not match", "danger");
            removeSpinner();
        } else {
            const userData = {
                email,
                name,
                username,
                password,
            };
            const verifyStartedReg = await props.startRegistration(userData);
            if (verifyStartedReg) {
                removeSpinner();
                const subject = "Verify your new CoScript account";
                const res = await sendEmail(email, subject, "registration");
                if (res)
                    localStorage.setItem("originalCode", res.data.originalCode);
            } else {
                removeSpinner();
            }
        }
    };

    /**
     * Uses the code stored in local storage to check if the code
     * the user entered is a match. If the code doesn't match, the
     * 'failed' state property is set to true and the 'verifyFailedMsg'
     * state property is set to the value sent back by the server
     *
     * @param {Object} e - event object
     */
    const onSubmitConfirm = async (e) => {
        e.preventDefault();
        const data = {
            originalCode: localStorage.getItem("originalCode"),
            confirmCode: confirmCode,
            token: props.regToken,
        };
        const res = await props.verifyEmail(data);
        if (res) {
            setFailed(true);
            setVerifyFailedMsg(res.msg);
        }
    };

    const { from } = props.location.state || {
        from: { route: "/" },
    };

    const { regUser, regToken } = props;
    const isLoggedIn = regUser ? regUser.isLoggedIn : false;
    const isRegistered = regUser ? regUser.isRegistered : false;

    if (isLoggedIn) {
        return <Redirect to={from} />;
    }

    if (isRegistered && !isLoggedIn) {
        return <Redirect to='/login' />;
    }

    if (!regToken) {
        return (
            <main className='register-body'>
                <NavBar />
                <RegistrationForm
                    email={email}
                    name={name}
                    username={username}
                    password={password}
                    confirmPassword={confirmPassword}
                    onChange={onChange}
                    onSubmit={onSubmit}
                    loading={loading}
                    regToken={props.regToken}
                    emailField={emailRef}
                />
            </main>
        );
    } else {
        return (
            <ConfirmRegistration
                isRegistered={isRegistered}
                onChangeConfirm={onChangeConfirm}
                onSubmitConfirm={onSubmitConfirm}
                verifyFailed={verifyRef}
            />
        );
    }
};

Register.propTypes = {
    user: PropTypes.object,
    regToken: PropTypes.string,
    displayAlert: PropTypes.func.isRequired,
    startRegistration: PropTypes.func.isRequired,
    verifyEmail: PropTypes.func.isRequired,
};

const setProps = (state) => ({
    user: state.auth.user,
    regUser: state.auth.regUser,
    regToken: state.auth.regToken,
});

export default connect(setProps, {
    displayAlert,
    startRegistration,
    verifyEmail,
})(withRouter(Register));
