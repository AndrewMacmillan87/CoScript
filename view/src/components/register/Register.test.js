import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter as Router, Link } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Provider } from "react-redux";
import { reduxStore } from "../../redux/store/reduxStore";

import { Register } from "./Register";
import NavBar from "../NavBar";
import ConfirmRegistration from "./ConfirmRegistration";
import PostButton from "../buttons/PostButton";

describe("Register component", () => {
    let props;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        props = {
            user: {},
            regToken: "",
            displayAlert: jest.fn(),
            startRegistration: jest.fn(),
            verifyEmail: jest.fn(),
            displayAlert: jest.fn(),
            location: {
                state: {},
            },
        };
    });

    afterEach(() => {
        props.regToken = "";
    });

    test("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(
                <Provider store={reduxStore}>
                    <Router>
                        <Register {...props} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render a main element with a className of register-body", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("main").length).toEqual(1);
        expect(wrapper.find({ className: "register-body" }).length).toEqual(1);
    });

    it("should render the NavBar component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(NavBar).length).toEqual(1);
    });

    it("should render the ConfirmRegistration component if regToken is truthy", () => {
        props.regToken = "123456789abcdefghijklmnopqrstuvwxyz";
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(ConfirmRegistration).length).toEqual(1);
    });
});

describe("RegistrationForm component", () => {
    let props, mockStartRegistration, mockVerifyEmail, mockDisplayAlert;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        mockStartRegistration = jest.fn();
        mockVerifyEmail = jest.fn();
        mockDisplayAlert = jest.fn();
        props = {
            user: {},
            regToken: "",
            displayAlert: jest.fn(),
            startRegistration: mockStartRegistration,
            verifyEmail: mockVerifyEmail,
            displayAlert: mockDisplayAlert,
            location: {
                state: {},
            },
        };
    });

    afterEach(() => {
        props.regToken = "";
    });

    it("should render a section element with a className of form-container", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("section").length).toEqual(1);
        expect(wrapper.find({ className: "form-container" }).length).toEqual(1);
    });

    it("should render a form element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("form").length).toEqual(1);
    });

    it("should render a h1 element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("h1").length).toEqual(1);
    });

    it("should render 5 label elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("label").length).toEqual(5);
    });

    it("should render 6 input elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("input").length).toEqual(6);
    });

    it("should render a p element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("p").length).toEqual(1);
    });

    it("should render a Link components", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Link).length).toEqual(1);
    });

    it("should render the PostButton component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(PostButton).length).toEqual(1);
    });

    it("should call `displayAlert()` if passwords do not match", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        wrapper
            .find("input")
            .at(0)
            .simulate("change", {
                target: { name: "email", value: "test@email.com" },
            });
        wrapper
            .find("input")
            .at(1)
            .simulate("change", {
                target: { name: "name", value: "Test Name" },
            });
        wrapper
            .find("input")
            .at(2)
            .simulate("change", {
                target: { name: "username", value: "test101" },
            });
        wrapper
            .find("input")
            .at(3)
            .simulate("change", {
                target: { name: "password", value: "password1234" },
            });
        wrapper
            .find("input")
            .at(4)
            .simulate("change", {
                target: { name: "confirmPassword", value: "password4321" },
            });
        wrapper.find("form").simulate("submit", { preventDefault: jest.fn() });
        expect(mockDisplayAlert).toBeCalled();
    });

    it("should call `startRegistration()` if passwords do match", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        wrapper
            .find("input")
            .at(0)
            .simulate("change", {
                target: { name: "email", value: "test@email.com" },
            });
        wrapper
            .find("input")
            .at(1)
            .simulate("change", {
                target: { name: "name", value: "Test Name" },
            });
        wrapper
            .find("input")
            .at(2)
            .simulate("change", {
                target: { name: "username", value: "test101" },
            });
        wrapper
            .find("input")
            .at(3)
            .simulate("change", {
                target: { name: "password", value: "password1234" },
            });
        wrapper
            .find("input")
            .at(4)
            .simulate("change", {
                target: { name: "confirmPassword", value: "password1234" },
            });
        wrapper.find("form").simulate("submit", { preventDefault: jest.fn() });
        expect(mockStartRegistration).toBeCalled();
    });
});

describe("ConfirmRegistration component", () => {
    let props, mockStartRegistration, mockVerifyEmail, mockDisplayAlert;
    beforeEach(() => {
        global.scrollTo = jest.fn();
        mockStartRegistration = jest.fn();
        mockVerifyEmail = jest.fn();
        mockDisplayAlert = jest.fn();
        props = {
            user: {},
            regToken: "123456789abcdefghijklmnopqrstuvwxyz",
            displayAlert: jest.fn(),
            startRegistration: mockStartRegistration,
            verifyEmail: mockVerifyEmail,
            displayAlert: mockDisplayAlert,
            location: {
                state: {},
            },
        };
    });

    it("should render a main element with a className of confirm-body", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("main").length).toEqual(1);
        expect(wrapper.find({ className: "confirm-body" }).length).toEqual(1);
    });

    it("should render a section element with a classNam of confirm-container", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("section").length).toEqual(1);
        expect(wrapper.find({ className: "confirm-container" }).length).toEqual(
            1
        );
    });

    it("should render 2 p tags", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("p").length).toEqual(2);
    });

    it("should render a form element", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("form").length).toEqual(1);
    });

    it("should render 2 input elements", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find("input").length).toEqual(2);
    });

    it("should render the PostButton component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(PostButton).length).toEqual(1);
    });

    it("should render a Link component", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        expect(wrapper.find(Link).length).toEqual(1);
    });

    it("should call `verifyEmail()` when the form is submitted", () => {
        const wrapper = mount(
            <Provider store={reduxStore}>
                <Router>
                    <Register {...props} />
                </Router>
            </Provider>
        );
        wrapper
            .find("input")
            .at(0)
            .simulate("change", {
                target: { name: "confirmCode", value: "123456" },
            });
        wrapper.find("form").simulate("submit", { preventDefault: jest.fn() });
        expect(mockVerifyEmail).toBeCalled();
    });
});
