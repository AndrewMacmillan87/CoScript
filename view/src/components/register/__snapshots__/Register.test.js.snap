// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders correctly 1`] = `
<main
  className="register-body"
>
  <nav
    className="navbar"
  />
  <section
    className="form-container"
  >
    <div
      className="registration-form"
    >
      <form
        onSubmit={[Function]}
      >
        <h1
          className="sign-up-header"
        >
          Sign Up
        </h1>
        <label
          className="reg-label"
          name="register-email"
        >
          Email*
        </label>
        <input
          className="reg-input"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email..."
          type="email"
          value=""
        />
        <label
          className="reg-label"
          name="register-name"
        >
          Name*
        </label>
        <input
          className="reg-input"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Name..."
          type="text"
          value=""
        />
        <label
          className="reg-label"
          name="register-username"
        >
          Username*
        </label>
        <input
          className="reg-input"
          name="username"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Username..."
          type="text"
          value=""
        />
        <label
          className="reg-label"
          name="register-password"
        >
          Password*
        </label>
        <input
          className="reg-input"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Password..."
          type="password"
          value=""
        />
        <label
          className="reg-label"
          name="confirm-password"
        >
          Confirm password*
        </label>
        <input
          className="reg-input"
          name="confirmPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Confirm password..."
          type="password"
          value=""
        />
        <br />
        <div
          className="form-links-container"
        >
          <p
            className="reg-link"
          >
            Already registered? Click
             
            <a
              href="/login"
              onClick={[Function]}
            >
              <span
                className="login-link"
              >
                here
              </span>
            </a>
          </p>
        </div>
        <div
          className="register-btn-container"
        >
          <input
            className="grn-btn"
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </section>
</main>
`;
